File: 프로그래머스/lv0/120956. 옹알이 （1）/옹알이 （1）.py
def check(word):
    p = {"a": "aya", "y": "ye", "w": "woo", "m": "ma"}
    i = 0
    while i < len(word):
        if word[i] not in p or word[i:i + len(p[word[i]])] != p[word[i]]:
            return 0
        i += len(p[word[i]]) if word[i] in p else 1
    return 1

def solution(babbling):
    answer = 0
    for w in babbling:
        answer += check(w)
    return answer

File: 프로그래머스/lv0/120956. 옹알이 （1）/옹알이 （1）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        String[] words = {"aya", "ye", "woo", "ma"};
        for (int i = 0; i < babbling.length; i++) {
            for (String w : words) {
                babbling[i] = babbling[i].replace(w, ".");
            }
            if (babbling[i].replace(".", "").length() == 0) {
                answer++;
            }
        }
        return answer;
    }
}